{
  final IgniteCache<String,Integer> cache=jcache().withAsync();
  cache.invoke("key2",ERR_PROCESSOR);
  assertThrows(log,new Callable<Object>(){
    @Override public Object call() throws Exception {
      IgniteFuture fut=cache.future().chain(new IgniteClosure<IgniteFuture,Object>(){
        @Override public Object apply(        IgniteFuture o){
          return o.get();
        }
      }
);
      fut.get();
      return null;
    }
  }
,EntryProcessorException.class,null);
}
